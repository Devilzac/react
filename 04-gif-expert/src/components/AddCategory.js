import React, {useState} from 'react';


export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("Hola mundo");

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        const text = inputValue.trim();
        if(text.length <= 1) return;
       //setCategories(cats => [...cats,inputValue]);
       onNewCategory(text);
       setInputValue('');
    }

  return (
    <form onSubmit={handleSubmit}>
        <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
        />
    </form>
  )
}
