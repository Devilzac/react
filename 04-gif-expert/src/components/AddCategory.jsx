import React, {useState} from 'react';
import PropTypes from 'prop-types'; 

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("Hola mundo");

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        const text = inputValue.trim();
        if(text.length <= 1) return;
       //setCategories(cats => [...cats,inputValue]);
       onNewCategory(text);
       setInputValue('');
    }

  return (
    <form onSubmit={handleSubmit}>
        <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
        />
    </form>
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}