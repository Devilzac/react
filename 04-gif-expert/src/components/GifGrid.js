import React from 'react'

export default function GifGrid({category}) {
const gifs =[1,2,3,4,5,6]

  return (
    <>
        <h3>{ category }</h3>
       {
        gifs.map(gif =>(
            <p key={gif}>gif</p>
            ))
       }
    </>
  )
}
