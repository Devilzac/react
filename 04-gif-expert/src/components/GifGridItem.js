import React from 'react'

export const GifGridItem = ({title, url,id}) => {
  return (
    <div className="card">
        <img width="400" height="200" src={url} alt={title}/>  
        <p>{title}</p>
    </div>

  )
}
