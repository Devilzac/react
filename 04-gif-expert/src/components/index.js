export * from './AddCategory';
export * from './GifGrid';
export * from './GifExpertApp';
export * from './GifGridItem';