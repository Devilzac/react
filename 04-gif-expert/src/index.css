* {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
}

.card-grid{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.card{
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin: 10px;
  
}

.card-grid img{
  object-fit: cover;
  width: 400px;
  height: 200px;
}