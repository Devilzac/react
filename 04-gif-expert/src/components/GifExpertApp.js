import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';

export const GifExpertApp = () => {
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    const onAddCategory = (newCategory)=>{
        if(categories.includes(newCategory)) return;
        setCategories([ newCategory, ...categories]);
    }

  return (
    <>
        <AddCategory 
            onNewCategory={ (value) => onAddCategory(value) }
        />
        <hr />
        {
            categories.map((category) =>(
                    <GifGrid 
                        key={ category }
                        category={ category } 
                    />
                ))
        }
        
    </>
  )
}
