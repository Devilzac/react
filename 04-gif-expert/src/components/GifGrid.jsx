import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';
//import { getGifs } from '../helpers/getGifs';

export function GifGrid({category}) {

  const { images, isLoading} = useFetchGifs(category);
  console.log({images, isLoading, category});



  return (
    <>
      <h3>{ category }</h3>
      {
        isLoading && (<h1>cargando...</h1>) 
      }
        <div className='card-grid'>
          {images.map( (images) =>(
            <GifGridItem key={images.id} 
              {...images}
            />
          ))}
        </div>
    </>
  )
}

GifGrid.propTypes= {
  category: PropTypes.string.isRequired,
}