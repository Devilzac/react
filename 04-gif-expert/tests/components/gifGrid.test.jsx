import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock('../../src/hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = "one punch";

    test('debe de mostrar el loading inicialmente', () => { 

        useFetchGifs.mockReturnValue({
            images:[],
            isLoading: true
        });

        render(<GifGrid category={ category }/>);
        expect( screen.getByText('cargando...'));
        expect( screen.getByText( category ) );
     });

     test('Debe de mostrar items cuando se cargan las imagenes mediante el useFetchGifs', () => { 
        const gifs = [
            {
                id: 'ABC',
                title: 'Saitama',
                url: 'https://localhost/saitama.jpg'
            },
            {
                id: 'CBA',
                title: 'goku',
                url: 'https://localhost/goku.jpg'
            }
        ];
        
        useFetchGifs.mockReturnValue({
            images:gifs,
            isLoading: false
        });

        
        
        const {container} = render(<GifGrid category={ category }/>);              
        expect(container).toMatchSnapshot();
        expect(screen.getAllByRole('img').length).toBe(2);
      })
});
