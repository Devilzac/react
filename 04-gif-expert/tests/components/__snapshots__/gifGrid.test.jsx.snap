// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan las imagenes mediante el useFetchGifs 1`] = `
<div>
  <h3>
    one punch
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="Saitama"
        height="200"
        src="https://localhost/saitama.jpg"
        width="400"
      />
      <p>
        Saitama
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="goku"
        height="200"
        src="https://localhost/goku.jpg"
        width="400"
      />
      <p>
        goku
      </p>
    </div>
  </div>
</div>
`;
